src/main/webapp/app/components/geneHistoryTooltip/gene-history-tooltip-utils.tsx

```diff
@@ -6,7 +6,9 @@ import { IDrug } from 'app/shared/model/drug.model';
 import { FlattenedHistory } from 'app/shared/util/firebase/firebase-history-utils';
 import DiffViewer from 'app/components/diff-viewer/DiffViewer';
 
-export default function constructTimeSeriesData(record: FlattenedHistory): RequiredTimeSeriesEventData | ExtraTimeSeriesEventData {
+export default function constructTimeSeriesData(
+  record: FlattenedHistory,
+): RequiredTimeSeriesEventData | ExtraTimeSeriesEventData | undefined {
```

Comment:
Now I wonder, should this method return undefined? @bprize15 when the record operation is unknown, it seems to me we should still return a event data.

Reply:
Based on my implementation from a while back, I handle an unknown operation by returning undefined and then parsing it out. This bit could probably use a refactor, but I think it makes sense given the current implementation.